### ISSUE TITLE
Postexec should be run after lazy deletes

### ISSUE THREAD

**Issue by ronisaacson** [rsnapshot/rsnapshot#99]
If you use preexec & postexec scripts to make the snapshot directory available (mount, toggle between read-only and read/write, etc.) and have lazy deletes turned on, deletes will always fail because the postexec runs before the deletes. When using lazy deletes, the postexec should not run until after the deletes are finished.

**Comment by bebehei** [comment]
Thanks, I will dig into this.

**Comment by sgpinkus** [comment]
Good point. Currently pre / post scripts run while the lock file is in place. Moving post script to after lazy delete would mean they'd run with the lock file not in place. This would w.h.p. break something for someone.
